lib/kubernetes-test-framework/src/wait_for_resource.rs

```diff
@@ -0,0 +1,92 @@
+//! Wait for a resource to reach a certain condition.
+
+use super::Result;
+use crate::util::run_command;
+use std::{ffi::OsStr, process::Stdio};
+use tokio::process::Command;
+
+/// Specify what condition to wait for.
+#[derive(Debug)]
+pub enum WaitFor<C>
+where
+    C: std::fmt::Display,
+{
+    /// Wait for resource deletion.
+    Delete,
+    /// Wait for the specified condition.
+    Condition(C),
+}
+
+/// Wait for a set of `resources` within a `namespace` to reach a `wait_for`
+/// condition.
+/// Use `extra` to pass additional arguments to `kubectl`.
+pub async fn namespace<CMD, NS, R, COND, EX>(
```

Comment:
I _think_ the convention for type parameters is `PascalCase` not all uppercase.

Reply:
I don't like those either. Pascal variant looks better here, I'll change it. Thx!